sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/LiteralGenerator.scala

```diff
@@ -175,10 +175,8 @@ object LiteralGenerator {
   }
 
   lazy val yearMonthIntervalLiteralGen: Gen[Literal] = {
-    for {
-      years <- Gen.choose(-178956969, 178956969)
-      months <- Gen.choose(-11, 11)
-    } yield Literal.create(Period.of(years, months, 0), YearMonthIntervalType)
+    for { months <- Gen.choose(-1 * maxIntervalInMonths, maxIntervalInMonths) }
+      yield Literal.create(Period.ofMonths(months), YearMonthIntervalType)
```

Comment:
I restricted the range of random year-month intervals here because such interval are used in the tests for expressions like `date -/+ interval` where `date` is random too. If we don't restrict the range, we will just get the overflow exception in most cases.

BTW, there is another random generator in `RandomDataGenerator` which doesn't have such restriction, and it can be used in other tests like saving/loading to datasources.